src/iter.rs

```diff
@@ -866,4 +989,493 @@ mod tests {
+            ("a", "aa"),             // Small alphabet range
+            ("a", "az"),             // Large alphabet range
+            ("m", "z"),              // Large alphabet range
+            ("apple", "applea"),     // Single element range with non-existent key
+            ("apple", "applez"),     // Overlapping range with close keys
+            ("kiwi", "kiwib"),       // Overlapping range with close keys
+        ];
+
+        for (start, end) in range_tests {
+            let range_start = VariableSizeKey::from_slice_with_termination(start.as_bytes());
+            let range_end = VariableSizeKey::from_slice_with_termination(end.as_bytes());
+
+            // Inclusive-Inclusive
+            let trie_results_incl_incl: Vec<_> = trie
+                .range(range_start.clone()..=range_end.clone())
+                .collect();
+            let btree_results_incl_incl: Vec<_> = btree
+                .range(start.as_bytes().to_vec()..=end.as_bytes().to_vec())
+                .map(|(k, v)| (k.clone(), *v))
+                .collect();
+            let trie_expected_incl_incl: Vec<_> = trie_results_incl_incl
+                .iter()
+                .map(|(k, v, _, _)| {
+                    let key_without_last_byte = &k[..k.len() - 1];
+                    (key_without_last_byte.to_vec(), **v)
+                })
+                .collect();
+            assert_eq!(
+                trie_expected_incl_incl, btree_results_incl_incl,
+                "Inclusive-Inclusive range scan from {} to {} failed",
+                start, end
+            );
+
+            // Inclusive-Exclusive
+            let trie_results_incl_excl: Vec<_> =
+                trie.range(range_start.clone()..range_end.clone()).collect();
+            let btree_results_incl_excl: Vec<_> = btree
+                .range(start.as_bytes().to_vec()..end.as_bytes().to_vec())
+                .map(|(k, v)| (k.clone(), *v))
+                .collect();
+            let trie_expected_incl_excl: Vec<_> = trie_results_incl_excl
+                .iter()
+                .map(|(k, v, _, _)| {
+                    let key_without_last_byte = &k[..k.len() - 1];
+                    (key_without_last_byte.to_vec(), **v)
+                })
+                .collect();
+            assert_eq!(
+                trie_expected_incl_excl, btree_results_incl_excl,
+                "Inclusive-Exclusive range scan from {} to {} failed",
+                start, end
+            );
+        }
+    }
+
+    // simulate an insert operation in surrealdb insert statement
+    fn setup_trie_and_btreemap() -> (Tree<VariableSizeKey, u16>, BTreeMap<VariableSizeKey, u16>) {
+        let mut tree: Tree<VariableSizeKey, u16> = Tree::<VariableSizeKey, u16>::new();
+        let mut map: BTreeMap<VariableSizeKey, u16> = BTreeMap::new();
+        let keys = vec![
+            VariableSizeKey::from_slice(&[47, 33, 110, 115, 116, 101, 115, 116, 0, 0]),
```

Comment:
Would be great not to use raw bytes here.

Reply:
Fixed!